rich/live.py

```diff
@@ -0,0 +1,347 @@
+        self.console = console if console is not None else get_console()
+        self._live_render = LiveRender(renderable)
+
+        self._redirect_stdout = redirect_stdout
+        self._redirect_stderr = redirect_stderr
+        self._restore_stdout: Optional[IO[str]] = None
+        self._restore_stderr: Optional[IO[str]] = None
+
+        self._lock = RLock()
+        self.ipy_widget: Optional[Any] = None
+        self.auto_refresh = auto_refresh
+        self._started: bool = False
+        self.transient = transient
+
+        self._refresh_thread: Optional[_RefreshThread] = None
+        self.refresh_per_second = refresh_per_second
+
+        self.hide_overflow = hide_overflow
+        self._hide_render = LiveRender("Terminal too small\n")
+        # cant store just clear_control as the live_render shape is lazily computed on render
+        self._clear_render = self._live_render
+
+    def start(self) -> None:
+        """Start live rendering display."""
+        with self._lock:
+            if self._started:
+                return
+
+            self.console.show_cursor(False)
+            self._enable_redirect_io()
+            self.console.push_render_hook(self)
+            self._started = True
+
+            if self.auto_refresh:
+                self._refresh_thread = _RefreshThread(self, self.refresh_per_second)
+                self._refresh_thread.start()
+
+    def stop(self) -> None:
+        """Stop live rendering display."""
+        with self._lock:
+            if not self._started:
+                return
+            self._started = False
+            try:
+                if self.auto_refresh and self._refresh_thread is not None:
+                    self._refresh_thread.stop()
+                self.refresh()
+                if self.console.is_terminal:
+                    self.console.line()
+            finally:
+                self._disable_redirect_io()
+                self.console.pop_render_hook()
+                self.console.show_cursor(True)
+
+            if self.transient:
+                self.console.control(self._live_render.restore_cursor())
+            if self.ipy_widget is not None and self.transient:  # pragma: no cover
+                self.ipy_widget.clear_output()
+                self.ipy_widget.close()
+
+    def __enter__(self) -> "Live":
+        self.start()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
+        self.stop()
+
+    def _enable_redirect_io(self):
+        """Enable redirecting of stdout / stderr."""
+        if self.console.is_terminal:
+            if self._redirect_stdout:
+                self._restore_stdout = sys.stdout
+                sys.stdout = _FileProxy(self.console, sys.stdout)
+            if self._redirect_stderr:
+                self._restore_stderr = sys.stderr
+                sys.stdout = _FileProxy(self.console, sys.stdout)
+
+    @property
+    def item(self) -> RenderableType:
```

Comment:
`item` may not be the best name here. I would think an item as in a dictionary items. How about simply "renderable" ?